yt/yt/server/controller_agent/controllers/operation_controller_detail.cpp

```diff
@@ -10035,10 +10035,26 @@ void TOperationControllerBase::InitUserJobSpecTemplate(
     jobSpec->set_rpc_proxy_worker_thread_pool_size(jobSpecConfig->RpcProxyWorkerThreadPoolSize);
 
     // Pass external docker image into job spec as is.
-    if (jobSpecConfig->DockerImage &&
-        !TDockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry).IsInternal())
-    {
-        jobSpec->set_docker_image(*jobSpecConfig->DockerImage);
+    if (jobSpecConfig->DockerImage) {
+        if (TDockerImageSpec(*jobSpecConfig->DockerImage, Config->DockerRegistry).IsInternal()) {
```

Comment:
For IsInternal it wouldn't work right now as is. When docker image is "internal" we doesn't pass it to job spec "set_docker_image" (below)

We need to rework how we handle internal images or invent something in-between.

I suppose it's better to handle mixed configuration where both docker_image and layers could be specified but exec node choose what it can handle. I have plans to use layers-artifacts as alternative way to deliver docker image.